### PR TITLE
Laboratory work #8, Zakirova Karina - 20FPL1

### PR DESCRIPTION
**Pull request by KarinaZakirova**

### CODE REVIEW

File: lab_7_llm/assets/index.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
```
Comment (SalnikovIgor) [inline lab_7_llm/assets/index.html]: Delete all 7 lab files from this PR

File: lab_8_llm/main.py [hunk 2]
```diff
@@ -43,6 +46,11 @@ def obtain(self) -> None:
         Raises:
             TypeError: In case of downloaded dataset is not pd.DataFrame
         """
+        self._raw_data = (load_dataset(self._hf_name, split='train')).to_pandas()
+
+    @property
+    def raw_data(self) -> DataFrame:
```
Comment (SalnikovIgor) [inline lab_8_llm/main.py]: The method has already been declared in the parent class `AbstractRawDataImporter`

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

File: lab_7_llm/assets/index.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+<html lang="en">
```
Comment (SalnikovIgor) [inline lab_7_llm/assets/index.html]: Checkout the files from 7 lab to the master

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Clean PR and fix tests

File: lab_8_llm/main.py (hunk 2 above)
Comment (KarinaZakirova) [reply lab_8_llm/main.py]: done

Comment (KarinaZakirova) [reply lab_8_llm/main.py]: sorry, not done
without this I can't pass mypy checks

File: lab_7_llm/assets/index.html (hunk 3 above)
Comment (KarinaZakirova) [reply lab_7_llm/assets/index.html]: done

File: lab_8_llm/main.py (hunk 2 above)
Comment (SalnikovIgor) [reply lab_8_llm/main.py]: Have the tests passed now?

File: lab_8_llm/assets/main.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        console.log(results_infer['infer'])
+        result.innerHTML = results_infer['infer'];
+}
+
+window.onload = function(){
+    const question = document.getElementById("question")
+    const button = document.getElementById("button")
+    const result = document.getElementById("result")
+    button.addEventListener("click", ()=>{sample_inference(question, result)})
+};
```
Comment (SalnikovIgor) [inline lab_8_llm/assets/main.js]: New line

File: lab_8_llm/assets/styles.css [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+.results>div{
+    width: 550px;
+    min-height: 47px;
+    height: auto;
+    background-color: var(--back_color);
+    padding: var(--padding_main);
+    border-bottom: var(--border);
+}
```
Comment (SalnikovIgor) [inline lab_8_llm/assets/styles.css]: New line

File: lab_8_llm/main.py (hunk 2 above)
Comment (SalnikovIgor) [reply lab_8_llm/main.py]: I did not see that these changes remained. Delete this method and variable and push it, let's look at the mypy error in the tests after update to last master. And in `start.py` need to add a check: `_raw_data` containt `None`? If yes, then rise error

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

**Pull request closed without merging**